include/remill/Arch/ArchBase.h

```diff
@@ -31,13 +31,27 @@ namespace remill {
 
 struct Register;
 
-// Internal base architecture for all Remill-internal architectures.
-class ArchBase : public remill::Arch {
- protected:
+
+/// A simple mixin for a default decoder with a null context and the default insdtruction lifter;
+class DefaultContextAndLifter : virtual public remill::Arch {
```

Comment:
`virtual` inheritance is a big hammer we may not need here - think how to make it without.